backend/tests/api/user_test.go

```diff
@@ -553,3 +553,308 @@ func TestCreateUserFailsOnMissingFields(t *testing.T) {
+
+	return tagIDs
+}
+
+func AssertUserTagsRespDB(app TestApp, assert *assert.A, resp *http.Response, id uuid.UUID) {
+	var respTags []models.Tag
+
+	// Retrieve the tags from the response:
+	err := json.NewDecoder(resp.Body).Decode(&respTags)
+
+	assert.NilError(err)
+
+	// Retrieve the user connected to the tags:
+	var dbUser models.User
+	err = app.Conn.First(&dbUser, id).Error
+
+	assert.NilError(err)
+
+	// Retrieve the tags in the bridge table associated with the user:
+	var dbTags []models.Tag
+	err = app.Conn.Model(&dbUser).Association("Tag").Find(&dbTags)
+
+	assert.NilError(err)
+
+	// Confirm all the resp tags are equal to the db tags:
+	for i, respTag := range respTags {
+		assert.Equal(respTag.ID, dbTags[i].ID)
+		assert.Equal(respTag.Name, dbTags[i].Name)
+		assert.Equal(respTag.CategoryID, dbTags[i].CategoryID)
+	}
+}
+
+func AssertSampleUserTagsRespDB(app TestApp, assert *assert.A, resp *http.Response, uuid uuid.UUID) {
+	AssertUserTagsRespDB(app, assert, resp, uuid)
+}
+
+func TestCreateUserTagsFailsOnInvalidDataType(t *testing.T) {
+	// Invalid tag data types:
+	invalidTags := []interface{}{
+		[]string{"1", "2", "34"},
+		[]models.Tag{{Name: "Test", CategoryID: uuid.UUID{}}, {Name: "Test2", CategoryID: uuid.UUID{}}},
+		[]float32{1.32, 23.5, 35.1},
+	}
+
+	// Test each of the invalid tags:
+	for _, tag := range invalidTags {
+		malformedTag := *SampleTagIDsFactory(nil)
+		malformedTag["tags"] = tag
+
+		TestRequest{
+			Method: fiber.MethodPost,
+			Path:   "/api/v1/users/1/tags/",
+			Body:   &malformedTag,
+		}.TestOnError(t, nil, errors.FailedToParseRequestBody).Close()
+	}
+}
+
+func TestCreateUserTagsFailsOnInvalidUserID(t *testing.T) {
+	badRequests := []string{
+		"0",
+		"-1",
+		"1.1",
+		"foo",
+		"null",
+	}
+
+	for _, badRequest := range badRequests {
+		TestRequest{
+			Method: fiber.MethodPost,
+			Path:   fmt.Sprintf("/api/v1/users/%s/tags", badRequest),
+			Body:   SampleTagIDsFactory(nil),
+		}.TestOnError(t, nil, errors.FailedToValidateID).Close()
+	}
+}
+
+type UUIDSlice []uuid.UUID
+
+var testUUID = uuid.New()
+
+func TestCreateUserTagsFailsOnInvalidKey(t *testing.T) {
+	appAssert, uuid := CreateSampleUser(t, nil)
+
+	invalidBody := []map[string]interface{}{
+		{
+			"tag": UUIDSlice{testUUID, testUUID},
+		},
+		{
+			"tagIDs": []uint{1, 2, 3},
+		},
+	}
+
+	for _, body := range invalidBody {
+		TestRequest{
+			Method: fiber.MethodPost,
+			Path:   fmt.Sprintf("/api/v1/users/%s/tags/", uuid),
+			Body:   &body,
+		}.TestOnError(t, &appAssert, errors.FailedToValidateUserTags)
+	}
+
+	appAssert.Close()
+}
+
+func TestCreateUserTagsFailsOnNonExistentUser(t *testing.T) {
+	TestRequest{
+		Method: fiber.MethodPost,
+		Path:   fmt.Sprintf("/api/v1/users/%s/tags", uuid.New()),
+		Body:   SampleTagIDsFactory(nil),
+	}.TestOnError(t, nil, errors.UserNotFound).Close()
+}
+
+func TestCreateUserTagsWorks(t *testing.T) {
+	appAssert, uuid := CreateSampleUser(t, nil)
+
+	// Confirm adding non-existent tags does nothing:
```

Comment:
does this belong in a separate test? this test seems to have 2 purpose statements